### ISSUE TITLE
nan predictions for unseen users and items

### ISSUE THREAD

**Issue by ankane** [david-cortes/cmfrec#21]
Hi, I wanted to get your thoughts on returning predictions instead of `nan` for unseen users and items with `predict`. I think it could return:

- unseen user and item: `global mean`
- unseen user: `global mean + item bias`
- unseen item: `global mean + user bias`

(could be an option for now to keep backward compatibility)

**Comment by david-cortes** [comment]
That's a good idea, that way it'd be more consistent with `predict_new` and `predict_cold`.

**Comment by ankane** [comment]
Awesome, thanks @david-cortes.

**Comment by ankane** [comment]
Hmm, still getting `nan` predictions with the Python and C libraries with 2.4.0.

**Comment by david-cortes** [comment]
You're right, just remembered that the python functions don't call the same public C functions at the moment.

But the C one should be working correctly - which function specifically are you calling?

Here's an example with the R version, which calls the same C function from the installable header:
```r
library(cmfrec)
set.seed(1)
X = matrix(rnorm(100), nrow=20)
m = CMF(X)
predict(m, 19:23, 48:52)
predict(m, 100:110, 100:110)
```
```
[1] 0.16197124 0.08319831 0.10888737 0.10888737 0.10888737
```
```
[1] 0.1088874 0.1088874 0.1088874 0.1088874 0.1088874 0.1088874 0.1088874 0.1088874 0.1088874 0.1088874 0.1088874
```

**Comment by ankane** [comment]
I'm calling `predict_multiple` like the Python library, so that's probably it. It looks like the R library calls `predict_X_old_collective_explicit`. Is that a better one to call?

**Comment by david-cortes** [comment]
Yes, and same one for the version called "new" and the topN equivalent. They are very short wrappers around the other C functions which are called from the python version, but do the extra processing in C instead.

**Comment by ankane** [comment]
Looks like that and 2.4.1 fixed it for the C library (2.4.1 isn't on PyPI fwiw). Thanks @david-cortes!
